File: Failed Ideas/communicator.cs
using System;
using System.IO;
using Chess_Input;

class Communicator : Program
{
    static Random rd = new Random(); // initialize rng
    public static void Main()
    {
        // function to handle and verify integer inputs
        static int ReadInt(int min = int.MinValue, int max = int.MaxValue, int userInt = 0)
        {
            while (true)
            {
                bool isValid = int.TryParse(Console.ReadLine(), out userInt);

                if (isValid == false) // if input is not an integer
                {
                    Console.WriteLine("Please input a valid integer.");
                }
                else if (userInt < min || userInt > max) // if input is too little or too large
                {
                    Console.WriteLine("Out of range, please input a valid number.");
                }
                else { break; } // if input is valid
            }

            return userInt;
        }

        static string colourSelect()
        {

            Console.WriteLine("Enter the value for the piece colour you want to play as:\n1 = White\n2 = Black\n3 = Random");
            int colour = ReadInt(1, 3);
            switch (colour)
            {
                case 1:
                    return "White";

                case 2:
                    return "Black";

                case 3:
                    int draw = rd.Next(1, 2);
                    if (draw == 1)
                    {
                        return "White";
                    }
                    else
                    {
                        return "Black";
                    }
                default:
                    return "ERROR: Invalid Input";
            }

        }

        static int eloSelect()
        {
            int elo;
            while (true)
            {
                Console.WriteLine("Enter your current Chess ELO (0 to 3000): ");
                string tempELO = Console.ReadLine();
                bool result = int.TryParse(tempELO, out elo);
    
                if (result == false || tempELO.Contains('-') || elo <= 0 || elo > 3000)
                {
                    Console.WriteLine("Please enter a valid integer from 0 to 3000.");
                }
                else
                {
                    return elo;
                }
            }
        }

        // gets user's elo and uses it to calculate bot elo
        static int difficultySelect()
        {
            Console.WriteLine("Enter your desired difficulty\n1 = Easy\n2 = Medium\n3 = Hard\n4 = NIGHTMARE...");
            int difficulty = ReadInt(1, 4);
    
            switch (difficulty)
            {
                case 1:
                    return eloSelect() / 2;
    
                case 2:
                    return eloSelect();
    
                case 3:
                    return eloSelect() * 2;
    
                case 4:
                    return 3700;
    
                default:
                    return 0;
            }
        }

        // store values in variables
        string colour = colourSelect();
        string botELO = difficultySelect().ToString();
        string playerMove = moveInput();

        // write to file
        File.WriteAllText("communication.txt", colour);
        File.AppendAllText("communication.txt", $" {botELO}");
        File.AppendAllText("communication.txt", $" {playerMove}");
        }
    }

File: Failed Ideas/communicator.py
# import libraries
import random
import gui

# initialize rng
random.seed()

# function to read and validate an integer input
def read_int(min_value=999999999, max_value=999999999, user_int=0):
  while True: # loop to continuously prompt user for an acceptable value
    try:
      user_int = int(input())
      if user_int < min_value or user_int > max_value:
        print("Out of range, please input a valid number.")
      else:
        break
    except ValueError:
      print("Please input a valid integer.")
  return user_int

# function to prompt user for piece color to play
def colour_select():
  print("Enter the value for the piece colour you want to play as:\n1 = White\n2 = Black\n3 = Random")
  colour = read_int(1, 3)
  if colour == 1:
    return "White"
  elif colour == 2:
    return "Black"
  elif colour == 3:
    return random.choice(["White", "Black"])
  else:
    return "ERROR: Invalid Input"


# function to prompt user for their elo
def elo_select():
  while True:
    try:
      temp_elo = input("Enter your current Chess ELO (0 to 3000): ")
      elo = int(temp_elo)
      if elo < 0 or elo > 3000:
        print("Please enter a valid integer from 0 to 3000.")
      else:
        return elo
    except ValueError: # if an incorrent value type is given, print following message
      print("Please enter a valid integer from 0 to 3000.")


# function to prompt user for a difficulty, then return the bot elo accordingly
def difficulty_select():
  print("Enter your desired difficulty\n1 = Easy\n2 = Medium\n3 = Hard\n4 = NIGHTMARE...")
  difficulty = read_int(1, 4)
  if difficulty == 1: # easy
    return elo_select() // 2
  elif difficulty == 2: # medium
    return elo_select()
  elif difficulty == 3: # hard
    return elo_select() * 2
  elif difficulty == 4: # nightmare
    return 3700
  else:
    pass

# store values into variables
colour = colour_select()
bot_elo = str(difficulty_select())
player_move = moveInput()  # using move input from gui program

# write values to txt file
with open("communication.txt", "w") as file:
  file.write(colour)
  file.write(f" {bot_elo}")
  file.write(f" {player_move}")
